packages/rrweb-worker/src/_image-bitmap-data-url-worker.ts

```diff
@@ -49,23 +51,40 @@ const worker: ImageBitmapDataURLResponseWorker = self;
 // eslint-disable-next-line @typescript-eslint/no-misused-promises
 worker.onmessage = async function (e) {
   if ('OffscreenCanvas' in globalThis) {
-    const { id, bitmap, width, height, dataURLOptions } = e.data;
+    const { id, bitmap, width, height, maxCanvasSize, dataURLOptions } = e.data;
 
     const transparentBase64 = getTransparentBlobFor(
       width,
       height,
       dataURLOptions,
     );
 
-    const offscreen = new OffscreenCanvas(width, height);
-    const ctx = offscreen.getContext('2d')!;
+    const [targetWidth, targetHeight] = getScaledDimensions(
+      width,
+      height,
+      maxCanvasSize,
+    );
+    const offscreen = new OffscreenCanvas(targetWidth, targetHeight);
+    const ctx = offscreen.getContext('bitmaprenderer')!;
+    const resizedBitmap =
+      targetWidth === width && targetHeight === height
+        ? bitmap
+        : // resize bitmap to fit within maxsize
```

Comment:
what will happen if this is not supported? 🤔

Reply:
I'm not sure tbh, it could throw or it could ignore the options.

Reply:
My previous commit did the resizing manually with `drawImage()`, but I'd rather use the builtin since it's pretty well-supported. I think throwing and not capturing exceptions and in the other case, if it doesn't throw and just ignores the options and captures the original size, are both acceptable outcomes.

I'm going to merge as-is, but let me know what you think and I can follow-up.